### PR TITLE
Remove default path settings

### PR DESCRIPTION
**Pull request by jasontedor**
This commit removes the default path settings for data and logs. With this change, we now ship the packages with these settings set in the elasticsearch.yml configuration file rather than going through the default.path.data and default.path.logs dance that we went through in the past.

Relates #25357

### CODE REVIEW

File: docs/reference/migration/migrate_6_0/packaging.asciidoc [hunk 1]
```diff
@@ -19,3 +19,13 @@ Elasticsearch should use another config file. Instead, `path.conf` is now a
 command-line flag. To start Elasticsearch with a custom config file, use `-c
 /path/to/config` or `--path.conf /path/to/config`. Here, `/path/to/config` is
 the *directory* containing the config file.
+
+==== Default path settings are removed
+
+Previous versions of Elasticsearch enabled setting `default.path.data` and
+`default.path.logs` so set the default data path and default logs path if they
```
Comment (rjernst) [inline docs/reference/migration/migrate_6_0/packaging.asciidoc]: typo: so -> to

File: distribution/rpm/src/main/packaging/init.d/elasticsearch [hunk 2]
```diff
@@ -35,8 +35,6 @@ fi
 ES_HOME="/usr/share/elasticsearch"
 MAX_OPEN_FILES=65536
 MAX_MAP_COUNT=262144
-LOG_DIR="/var/log/elasticsearch"
```
Comment (rjernst) [inline distribution/rpm/src/main/packaging/init.d/elasticsearch]: Should we fail hard if these env vars are set? Not sure where the appropriate place to check it is. Somewhere in the service files?

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by jasontedor** [comment]
retest this please

File: distribution/rpm/src/main/packaging/init.d/elasticsearch (hunk 2 above)
Comment (jasontedor) [inline distribution/rpm/src/main/packaging/init.d/elasticsearch]: That would be the only place, these environment variables were only ever used in the service files and were never exported from them. Let me do this as a follow up.

**Pull request merged**